### PR TITLE
Add documentation about extending website search

### PR DESCRIPTION
**Pull request by luca-rath**
| Q | A
| --- | ---
| Fixed tickets | fixes #issuenum
| Related PRs | sulu/sulu#5460
| License | MIT

#### What's in this PR?

Add documentation about extending website search.

#### Why?

Because it should be documented.

### CODE REVIEW

File: cookbook/extending-website-search.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+The configuration looks like the following:
+
+.. code-block:: yaml
+
+    # config/packages/sulu_search.yaml
+
+    sulu_search:
+        website:
+            indexes:
+                articles: article_published
```
Comment (danrot) [inline cookbook/extending-website-search.rst:19]: Shouldn't the articles be preprended by the `SuluArticleBundle`? Why should we burden the developer with doing this on their own?

Comment (danrot) [reply cookbook/extending-website-search.rst:19]: Personally I would consider this a bug fix, not a BC break, therefore I would not care about the changed behavior.

**Comment by alexander-schranz** [comment]
I personally worried that the index will not take the webspace settings into account and the search links will end in a 404?

@wachterjohannes or is this handled somewhere?

**Comment by danrot** [comment]
Well, if that is the case we would have to fix that anyway, wouldn't we?

File: cookbook/extending-website-search.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
+Extending website search
```
Comment (alexander-schranz) [inline cookbook/extending-website-search.rst:1]: Not sure about the place should this not be in the `SuluSearchBundle`?

Comment (danrot) [reply cookbook/extending-website-search.rst:1]: Good point, that file does even already exist: https://github.com/sulu/sulu-docs/blob/master/bundles/search.rst

**Comment by wachterjohannes** [comment]
@alexander-schranz i am not sure that would have to be checked

**Review by alexander-schranz — APPROVED** [review]
Looks good from my side.

File: bundles/search.rst [hunk 3]
```diff
@@ -53,6 +53,52 @@ So a sample configuration would look like this:
...
+An example for this would be the index for the pages:
+
+.. code-block:: yaml
+
+    # config/packages/sulu_search.yaml
+
+    sulu_search:
+        website:
+            indexes:
+                pages: page_#webspace#_published
```
Comment (danrot) [inline bundles/search.rst:100]: Shouldn't we use another example? If people find that, they might think they have to do this, although I would say they absolutely should not... Maybe just use something like `example_#published#`?

**Review by danrot — APPROVED** [review]

**Pull request merged**